actor/kvcounter/src/lib.rs

```diff
@@ -0,0 +1,46 @@
+use serde_json::json;
+use wasmbus_rpc::actor::prelude::*;
+use wasmcloud_interface_httpserver::{HttpRequest, HttpResponse, HttpServer, HttpServerReceiver};
+use wasmcloud_interface_keyvalue::{IncrementRequest, KeyValue, KeyValueSender};
+
+#[derive(Debug, Default, Actor, HealthResponder)]
+#[services(Actor, HttpServer)]
+struct KvCounterActor {}
+
+/// Implementation of HttpServer trait methods
+#[async_trait]
+impl HttpServer for KvCounterActor {
+    async fn handle_request(&self, ctx: &Context, req: &HttpRequest) -> RpcResult<HttpResponse> {
+        // make friendlier key
+        let key = format!("counter:{}", req.path.replace('/', ":"));
+
+        // bonus: use specified amount from query, or 1
+        let amount: i32 = form_urlencoded::parse(req.query_string.as_bytes())
+            .find(|(n, _)| n == "amount")
+            .map(|(_, v)| v.parse::<i32>())
+            .unwrap_or(Ok(1))
+            .unwrap_or(1);
+
+        // increment the value in kv and send response in json
+        let (body, status_code) = match increment_counter(ctx, key, amount).await {
+            Ok(v) => (json!({ "counter": v }).to_string(), 200),
+            // if we caught an error, return it to client
+            Err(e) => (json!({ "error": e.to_string() }).to_string(), 500),
+        };
+        let resp = HttpResponse {
+            body: body.as_bytes().to_vec(),
+            status_code,
+            ..Default::default()
+        };
+        Ok(resp)
+    }
+}
+
+/// increment the counter by the amount, returning the new value
+async fn increment_counter(ctx: &Context, key: String, value: i32) -> RpcResult<i32> {
+    let tx = WasmHost::to_provider("wasmcloud:keyvalue", "default")?;
```

Comment:
The 0.18 and earlier versions of the actor SDKs all made the default case the easiest. If you're only ever using the `default` link name, then you should simply be able to call `KeyValueSender::new()` and it should assume the use of `default`. We should only have to call `WasmHost::to_provider("...", "...") if we want to change the link name. 

Further, requiring a developer to manually type the contract ID shouldn't be required, this should be made available to them as a constant in the crate they rely on for the key-value interface. `KeyValueSender` comes from the previously generated library, so it should already know its contract ID and not put the burden on the developer for typing that.

Reply:
fixed

Reply:
This doesn't appear to be fixed... you're still using the magic string "wasmcloud:keyvalue" and supplying "default" explicitly when, if you don't need anything other than the default, you shouldn't have to supply one (worse case scenario make people supply `None`, best case, the `to_provider` function doesn't take a link name, and you have a `to_provider_with_linkname` that takes the link name.

Reply:
What if we have the following signatures:
* `WasmHost::to_provider(keyvalue::CONTRACT_ID)` (link name is "default")
* `WasmHost::to_provider_with_linkname(keyvalue::CONTRACT_ID, "florbleghast")` (explicit link name)